src/Base/Rotation.h

```diff
@@ -112,6 +92,33 @@ class BaseExport Rotation
 
         EulerSequenceLast,
     };
+
+    /// Utility function to create Rotation based on direction / normal vector
+    static Rotation fromNormalVector(const Vector3d& normal);

```

Comment:
This function defines the normal vector to be Z globally. I would prefer if Rotation doesn’t do assumptions like this.

Would it be an option to create a const normalVector{0, 0, 0} in Attacher.cpp, that can be used in the two places instead of the new function?

Reply:
I'd argue that this is assumption that should be made globally. If we are dealing with normal vectors they should be always represented by one axis vector in the whole application. I don't see a good reason why it should not be global and I see problems that can happen when one part of application assumes Z to be the normal vector and another assumes let's say Y.

Reply:
In that case, I think it should be explicitly defined somewhere rather than having it only as an assumption in this function.

Reply:
I can update the doc comment, it seems like reasonable place for such information. 

In general I feel that in 3D space it is assumed that Z axis defines the normal vector for surfaces. It is quite natural because surfaces are 2D objects and so have only XY (or UV to differentiate surface coordinates from 3D space) space to work with and so the normal vector, which is perpendicular to both must be Z. Given that reasoning I could update the name to `fromSurfaceNormalVector` which should be more clear.

Assembly uses the same assumption and I bet that we would find more such assumptions across the codebase